src/nbl/video/COpenGLRenderpassIndependentPipeline.cpp

```diff
@@ -0,0 +1,33 @@
+#include "nbl/video/COpenGLRenderpassIndependentPipeline.h"
+
+#ifdef _NBL_COMPILE_WITH_OPENGL_
+
+#include "nbl/video/IOpenGL_LogicalDevice.h"
+
+namespace nbl {
+namespace video
+{
+
+COpenGLRenderpassIndependentPipeline::~COpenGLRenderpassIndependentPipeline()
+{
+    constexpr uint32_t MaxNamesPerStage = 128u;
+    const auto namesPerStage = m_GLprograms->size() / SHADER_STAGE_COUNT;
+    assert(namesPerStage < MaxNamesPerStage);
+
+    GLuint names[MaxNamesPerStage]{};
+    for (uint32_t i = 0u; i < SHADER_STAGE_COUNT; ++i)
+    {
+        if (!getShaderAtIndex(i))
+            continue;
+
+        for (uint32_t ctxid = 0u; ctxid < namesPerStage; ++ctxid)
+            names[ctxid] = getShaderGLnameForCtx(i, ctxid);
+
+        m_device->destroySpecializedShader(namesPerStage, names);
+    }
+}
```

Comment:
why does the pipeline destroy the shaders?

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/28#discussion_r582284021